packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/edge/EdgeLayout.ts

```diff
@@ -40,6 +47,64 @@ export const getUpdatedConnectionHandles: GetUpdatedConnectionHandlesParameters
     }
   );
 
+  const sidedHandlesIds = getHandlesIdsFromNode(sourceNode);
+  const otherEndsCenter: Map<string, XYPosition> = new Map<string, XYPosition>();
+
+  edges.forEach((edge) => {
+    if (edge.sourceNode && edge.targetNode && edge.sourceHandle && edge.targetHandle) {
+      if (sidedHandlesIds.find((id) => id === edge.sourceHandle)) {
+        let node = _nodes.find((n) => n.id === edge.targetNode?.id);
+        if (node) {
+          otherEndsCenter.set(edge.sourceHandle, getNodeCenter(node, _nodes));
+        }
+      }
+      if (sidedHandlesIds.find((id) => id === edge.targetHandle)) {
+        let node = _nodes.find((n) => n.id === edge.sourceNode?.id);
+        if (node) {
+          otherEndsCenter.set(edge.targetHandle, getNodeCenter(node, _nodes));
+        }
+      }
+    }
+  });
+
+  sourceNode.data.connectionHandles
+    .filter((handle) => !handle.virtual)
+    .map((nodeConnectionHandle: ConnectionHandle) => {
```

Comment:
This `map` should be a `forEach`: we do not actually care about the return value, we just want to iterate on each handle and mutate them in-place.